wcivf/apps/elections/views/postcode_view.py

```diff
@@ -80,6 +82,29 @@ def get_todays_ballots(self):
             election__election_date=timezone.now().date()
         )
 
+    def multiple_city_of_london_elections_today(self):
+        """
+        Checks if there are multiple elections taking place today in the City
+        of London. This is used to determine if it is safe to display polling
+        station open/close times in the template. As if there are multiple then
+        it is unclear what time the polls would be open. See this issue for
+        more info https://github.com/DemocracyClub/WhoCanIVoteFor/issues/441
+        """
+        ballots = self.get_todays_ballots()
```

Comment:
Just an idea on this one - both `get_todays_ballots()` and `ballot_paper_id__startswith="local.city-of-london"` will cause a DB query per request, just to catch an edge case that might never happen.

I think it might be better to change the return value of `get_todays_ballots()` (or maybe allow passing in `flat=True`) to just return a set of ballot paper IDs.

This would mean we only run a single query, but we can also do `len()` rather than `count()` on the set and then use python to work out if there's a city of London ballot in the set.

Another option that would prevent adding another query at all would be to annotate another field like we do here: https://github.com/DemocracyClub/WhoCanIVoteFor/blob/99484efa37e5a2f1b0fe49f60e2983d1956900c1/wcivf/apps/elections/views/mixins.py#L62

This would look like:
```py
pes = pes.annotate(
            is_today=Case(
                When(election__election_date=date.today(), then=1),
                default=0,
                output_field=IntegerField(),
            )
        )
```

`get_todays_ballots` could then be just `return [b for ballot in self.ballots if b.is_today]` and `multiple_city_of_london_elections_today` could just iterate over the list.

Reply:
Yeah you are totally right about the extra DB queries - should have thought more about the performance implications. I liked the `annote` idea, but then remembered I already added the `is_election_day` method on the Election model, so thought I could use that to do the same thing. And then updated `multiple_city_of_london_elections_today` accordingly to work with a list instead of a QS. Tests still pass and think this still results in removing the additional DB queries

Reply:
Nice!